tests/integration/ha_tests/test_smoke.py

```diff
@@ -0,0 +1,218 @@
+
+import pytest
+from pytest_operator.plugin import OpsTest
+from tenacity import Retrying, stop_after_delay, wait_fixed
+
+from ..helpers import (
+    CHARM_SERIES,
+    DATABASE_APP_NAME,
+    scale_application,
+)
+from .helpers import (
+    apply_pvc_config,
+    change_pv_reclaim_policy,
+    change_pvc_pv_name,
+    check_db,
+    check_system_id_mismatch,
+    create_db,
+    delete_pvc,
+    get_any_deatached_storage,
+    get_pv,
+    get_pvc,
+    get_storage_id,
+    is_postgresql_ready,
+    is_storage_exists,
+    remove_pv_claimref,
+    remove_unit_force,
+)
+
+TEST_DATABASE_RELATION_NAME = "test_database"
+DUP_DATABASE_APP_NAME = DATABASE_APP_NAME + "2"
+
+logger = logging.getLogger(__name__)
+
+env = os.environ
+env["KUBECONFIG"] = os.path.expanduser("~/.kube/config")
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_app_force_removal(ops_test: OpsTest):
+    """Remove unit with force while storage is alive."""
+    global primary_pv, primary_pvc
+    # Deploy the charm.
+    async with ops_test.fast_forward():
+        await ops_test.model.deploy(
+            DATABASE_APP_NAME,
+            application_name=DATABASE_APP_NAME,
+            num_units=1,
+            channel="14/stable",
+            series=CHARM_SERIES,
+            trust=True,
+            config={"profile": "testing"},
+        )
+
+        await ops_test.model.wait_for_idle(status="active", timeout=1000)
+
+        assert ops_test.model.applications[DATABASE_APP_NAME].units[0].workload_status == "active"
+
+        primary_name = ops_test.model.applications[DATABASE_APP_NAME].units[0].name
+
+        logger.info("waiting for postgresql")
+        for attempt in Retrying(stop=stop_after_delay(15 * 3), wait=wait_fixed(3), reraise=True):
+            with attempt:
+                assert await is_postgresql_ready(ops_test, primary_name)
+
+        # Create test database to check there is no resources conflicts
+        logger.info("creating db")
+        await create_db(ops_test, DATABASE_APP_NAME, TEST_DATABASE_RELATION_NAME)
+
+        assert primary_name
+
+        logger.info(f"get pvc for {primary_name}")
+        primary_pvc = get_pvc(ops_test, primary_name)
+
+        assert primary_pvc
+
+        logger.info(f"get pv for {primary_name}")
+        primary_pv = get_pv(ops_test, primary_name)
+
+        assert primary_pv
+
+        logger.info("get storage id")
+        storage_id = await get_storage_id(ops_test, primary_name)
+
+        assert storage_id
+
+        # Force remove unit without storage removal
+        logger.info("scale to 0 with force")
+        await remove_unit_force(ops_test, 1)
+
+        # Storage will remain with deatached status
+        logger.info("werifing is storage exists")
+        for attempt in Retrying(stop=stop_after_delay(15 * 3), wait=wait_fixed(3), reraise=True):
+            with attempt:
+                assert await is_storage_exists(ops_test, storage_id, include_detached=True)
```

Comment:
`include_detached` was removed from the helper function, but it's still here.

Reply:
Removed it